File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        current_sum = nums[0]
        maximum = nums[0]

        for i in range(1, len(nums)):
            current_sum = max(nums[i], current_sum + nums[i])
            maximum = max(maximum, current_sum)

        return maximum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int s=nums[0];
        int max_s=nums[0];
        for(int i=1;i<nums.length;i++){
            s=Math.max(nums[i],s+nums[i]);
            max_s=Math.max(max_s,s);
        }
        return max_s;
    }
}